jobs/orcid/ReconcileOrcidReviewPutCode.php

```diff
@@ -0,0 +1,126 @@
+                    $ojsReviews = [];
+
+                    foreach ($data['group'] as $dataGroup) {
+                        // Select only peer reviews submitted from OJS
+                        foreach ($dataGroup['peer-review-group'] as $peerReview) {
+                            if ($peerReview['peer-review-summary'][0]['source']['source-client-id']['path'] === $context->getData(OrcidManager::CLIENT_ID)) {
+                                $ojsReviews[] = $peerReview;
+                            }
+                        }
+                    }
+
+                    foreach ($ojsReviews as $review) {
+                        $reviewSummary = $review['peer-review-summary'][0];
+                        $externalIds = $reviewSummary['external-ids']['external-id'];
+                        $putCode = $reviewSummary['put-code'];
+                        $reviewRoundId = null;
+
+                        foreach ($externalIds as $externalId) {
+                            if ($externalId['external-id-type'] === 'source-work-id') {
+                                $reviewRoundId = $externalId['external-id-value'];
+                                break;
+                            }
+                        }
+
+                        if (!$reviewRoundId) {
+                            continue;
+                        }
+
+                        OrcidManager::logInfo("Processing previously deposited review with put-code: {$putCode}, and external-id-value (reviewRoundId) {$reviewRoundId}.");
+
+                        // Find review assignment and update put-code
+                        $reviewAssignment = Repo::reviewAssignment()
+                            ->getCollector()
+                            ->filterByContextIds([$this->contextId])
+                            ->filterByReviewerIds([$reviewer->getId()])
+                            ->filterByReviewRoundIds([$reviewRoundId])
+                            ->getMany()->first(); /**@var ReviewAssignment $reviewAssignment * */
+
+                        if ($reviewAssignment) {
+                            $reviewAssignment->setData('orcidReviewPutCode', $putCode);
+                            Repo::reviewAssignment()->edit($reviewAssignment, ['orcidReviewPutCode']);
+                            OrcidManager::logInfo("Successfully updated put-code ({$putCode}) for review assignment with ID {$reviewAssignment->getId()}, reviewRoundId {$reviewRoundId}, and userID {$reviewer->getId()}.");
+                        } else {
+                            OrcidManager::logInfo("Did not find a review assignment for deposited review with put-code {$putCode}, and external-id-value (reviewRoundId) {$reviewRoundId}.");
+                        }
+                    }
+
+                    // Remove old put-code from user object
+                    $reviewer->setData('orcidReviewPutCode', null);
+                    Repo::user()->edit($reviewer, ['orcidReviewPutCode']);
```

Comment:
I did a cursory look, but it seems like this isn't actually deleting the `orcidReviewPutCode` from the user settings table. Could you have a look?

Reply:
I've noticed this issue elsewhere and have opened a issue with PR for it. Do you mind taking a look?
https://github.com/pkp/pkp-lib/issues/10868

Reply:
Done!